src/clients/authorization/AuthorizationExport.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  FormGroup,
+  PageSection,
+  ActionGroup,
+  Button,
+  TextArea,
+  AlertVariant,
+} from "@patternfly/react-core";
+
+import { FormAccess } from "../../components/form-access/FormAccess";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { useParams } from "react-router-dom";
+import FileSaver from "file-saver";
+import { prettyPrintJSON } from "../../util";
+import "./authorization-details.css";
+import { useAlerts } from "../../components/alert/Alerts";
+
+export const AuthorizationExport = () => {
+  const { t } = useTranslation("clients");
+  const adminClient = useAdminClient();
+  const { clientId } = useParams<{ clientId: string }>();
+  const { addAlert, addError } = useAlerts();
+
+  const [code, setCode] = useState<string>();
+  const [authorizationDetails, setAuthorizationDetails] = useState("");
+
+  useFetch(
+    async () =>
```

Comment:
The function already returns a Promise, so there is no need to mark this as `async`.
```suggestion
    () =>
```